### PR TITLE
Add ruff linter

### PR DESCRIPTION
**Pull request by luca-medeiros**
Added ruff linter to pyproject with some basic configs after seeing it on the roadmap. I think we could find the best settings for this project later on.

Also ran ruff format on the current project with auto-fix on.

### CODE REVIEW

File: src/migrations/env.py [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
+from app.core.db.database import Base
 from sqlalchemy import pool
 from sqlalchemy.engine import Connection
 from sqlalchemy.ext.asyncio import async_engine_from_config
 
-from alembic import context
-
-from app.core.db.database import Base
-from app.core.config import settings
-from app.models.user import *
```
Comment (luca-medeiros) [inline src/migrations/env.py:12]: @igorbenav mind checking this change? not sure what we are importing here and ruff doesn't allow star imports

**Comment by luca-medeiros** [comment]
I could add a pre-commit config as well

**Comment by igorbenav** [comment]
Nice one, @luca-medeiros! I'll check if this extra import is necessary in alembic migrations, but I don't think it is since I'm importing Base. No Idea why I created it in the first place

File: src/migrations/env.py (hunk 1 above)
Comment (igorbenav) [reply src/migrations/env.py:12]: It's working fine

**Review by igorbenav — APPROVED** [review]

**Comment by igorbenav** [comment]
@luca-medeiros if you'd like to add the pre-commit config, I'd appreciate it a lot

**Pull request merged**